builtin/mainmenu/dlg_mg_flags.lua

```diff
@@ -0,0 +1,243 @@
+--Minetest
+--Copyright (C) 2016 srifqi, Muhammad Rifqi Priyo Susanto
+--
+--This program is free software; you can redistribute it and/or modify
+--it under the terms of the GNU Lesser General Public License as published by
+--the Free Software Foundation; either version 2.1 of the License, or
+--(at your option) any later version.
+--
+--This program is distributed in the hope that it will be useful,
+--but WITHOUT ANY WARRANTY; without even the implied warranty of
+--MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+--GNU Lesser General Public License for more details.
+--
+--You should have received a copy of the GNU Lesser General Public License along
+--with this program; if not, write to the Free Software Foundation, Inc.,
+--51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+local mg_first_time = true
+
+-- Use default if not defined
+
+-- Water level
+local water_level = 1
+
+-- General attributes
+local flag_list_g = {
+	caves		= true,
+	dungeons	= true,
+	light		= true,
+	decorations	= true
+}
+
+-- v6 mapgen
+local flag_list_6 = {
+	biomeblend	= true,
+	mudflow		= true,
+	snowbiomes	= true,
+	jungles		= true,
+	flat		= false,
+	trees		= true
+}
+
+-- v7 mapgen
+local flag_list_7 = {
+	mountains	= true,
+	ridges		= true
+}
+
+-- flat mapgen
+local flag_list_f = {
+	lakes		= false,
+	hills		= false
+}
+
+-- valleys mapgen
+local flag_list_v = {
+	altitude_chill	= true,
+	humid_rivers	= true
+}
+
+local function update_mg_flags()
+	local function mg_flags_helper(flag_list)
+		local first = true
+		local flag_str = ""
+		for flag, val in pairs(flag_list) do
+			flag_str = flag_str .. (first and "" or ", ") .. (val == true and "" or "no") .. flag
+			first = false
+		end
+		return flag_str
+	end
+	
+	core.setting_set("mg_flags", mg_flags_helper(flag_list_g))
+	core.setting_set("mgv6_spflags", mg_flags_helper(flag_list_6))
+	core.setting_set("mgv7_spflags", mg_flags_helper(flag_list_7))
+	core.setting_set("mgflat_spflags", mg_flags_helper(flag_list_f))
+	core.setting_set("mg_valleys_spflags", mg_flags_helper(flag_list_v))
+	core.setting_set("water_level", water_level)
+end
+
+-- Load default settings if not exist
+if core.setting_get("water_level") == nil then
+	core.setting_set("water_level", 1)
```

Comment:
No need to set the defaults or checking for the existence. Please correct me when I see this one the wrong way but the default value is loaded if there's one defined in `defaultsettings.cpp`.